lib/mongodb.js

```diff
@@ -1798,6 +1803,23 @@ MongoDB.prototype.isObjectIDProperty = function(model, prop, value) {
   }
 };
 
+function sanitizeFilter(filter, options) {
+  options = Object.assign({}, options);
+  if (options && options.disableSanitization) return filter;
+  if (!filter || typeof filter !== 'object') return filter;
+
+  for (const key in filter) {
+    if (key === '$where' || key === 'mapReduce') {
```

Comment:
`$mapReduce`?

Reply:
According to the docs it's `mapReduce`. 

https://docs.mongodb.com/manual/core/server-side-javascript/

Reply:
Can the special keys nested in the filter deeper than the 1st level?

Reply:
Nope. As pointed out by the test case below that @shimks commented on, Mongo only executes these at the top most level and not at the property level.